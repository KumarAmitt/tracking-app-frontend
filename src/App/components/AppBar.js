import React from 'react';

const AppBar = ({ title }) => (
  <>
    <h2>{title}</h2>
  </>
);

export default AppBar;
