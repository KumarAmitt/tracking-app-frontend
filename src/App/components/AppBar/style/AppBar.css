.appbar {
  background-color: var(--accent-primary);
  display: flex;
  padding: 1.2em 1em 0.6em;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 10;
}

.appbar .title {
  font-weight: bold;
  font-size: 1.1rem;
  color: #eee;
}

.appbar a {
  color: #eee;
  position: absolute;
  left: 0.8em;
}
