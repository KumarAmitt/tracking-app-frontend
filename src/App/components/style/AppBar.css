.appbar {
  background-color: var(--accent-primary);
  display: flex;
  padding: 1.2em 1em 0.6em;
  justify-content: center;
}

.appbar .title {
  font-weight: bold;
  font-size: 1.1rem;
  color: #eee;
}

.appbar a {
  color: #eee;
  position: absolute;
  left: 0.8em;
}
