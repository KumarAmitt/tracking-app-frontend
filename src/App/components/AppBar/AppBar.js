import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './style/AppBar.css';
import ArrowBackIosOutlinedIcon from '@material-ui/icons/ArrowBackIosOutlined';

const AppBar = ({ title, link }) => (
  <>
    <div className="appbar">
      <Link to={link}>
        <ArrowBackIosOutlinedIcon />
      </Link>
      <div className="title">{title}</div>
    </div>
  </>
);

AppBar.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.string,
};

AppBar.defaultProps = {
  link: '/add_deal',
};

export default AppBar;
