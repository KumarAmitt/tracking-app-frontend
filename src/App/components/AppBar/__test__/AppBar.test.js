import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import AppBar from '../AppBar';

const store = configureAppStore();

describe('AppBar', () => {
  test('should match the snapshot of AppBar', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <AppBar title="appbar" />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
