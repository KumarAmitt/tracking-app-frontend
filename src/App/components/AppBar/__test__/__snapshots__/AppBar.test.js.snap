// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should match the snapshot of AppBar 1`] = `
<div
  className="appbar"
>
  <a
    href="/add_deal"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
      />
    </svg>
  </a>
  <div
    className="title"
  >
    appbar
  </div>
</div>
`;
